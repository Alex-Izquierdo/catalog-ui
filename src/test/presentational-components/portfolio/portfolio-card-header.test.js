import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';

import PortfolioCardHeader from '../../../presentational-components/portfolio/portfolio-card-header';

describe('<PortfolioCardHeader />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      portfolioName: 'foo'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfolioCardHeader { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render actions and react on user events', () => {
    const actionClick = jest.fn();
    const actions = [
      <button key="foo" onClick={ () => actionClick('foo') }></button>
    ];
    const wrapper = mount(
      <MemoryRouter>
        <PortfolioCardHeader { ...initialProps } headerActions={ actions } />
      </MemoryRouter>
    );

    const buttons = wrapper.find('button');
    expect(buttons).toHaveLength(1);
    buttons.simulate('click');
    expect(actionClick).toHaveBeenCalledWith('foo');
  });
});
