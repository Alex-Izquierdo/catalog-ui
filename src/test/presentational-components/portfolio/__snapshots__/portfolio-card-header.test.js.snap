// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioCardHeader /> should render correctly 1`] = `
<div
  className="portfolio-card-header"
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
    }
  }
>
  <Toolbar
    className={null}
  >
    <ToolbarGroup
      className={null}
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "auto",
          "paddingBottom": 16,
        }
      }
    />
  </Toolbar>
  <Text
    className="elipsis-text-overflow"
    component="h4"
  >
    foo
  </Text>
</div>
`;
